### PR TITLE
llvm 6.0 and libFuzzer implementation

### PR DESCRIPTION
**Pull request by theopolis**
This adds a new target (optional) `osqueryf`. This is built when using `make libfuzz`.

To use `libfuzzer` we need LLVM 6.0 so this also includes an update to the dependencies for `llvm` and `libcpp`. The `libcpp` dependency is what contains libFuzzer in this incarnation. This allows us to build libFuzzer with a clang that does not contain it, since libFuzzer needs a c++ runtime.

This is what found 6+ RapidJSON problems with config and config parsers. I am sure there are other fund parts you can fuzz by swapping out the content of `fuzz.cpp`.

### CODE REVIEW

**Comment by osqueryer** [comment]
:no_good: The commit bec764fdf70c662686a60030f3cf490d05786cf3 (Job results: [3513](https://jenkins.osquery.io/job/osqueryPullRequestAudit/3513/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:-1: The commit ce082afc4aa7085e24a929a7ea50897828d1d720 (Job results: [7046](https://jenkins.osquery.io/job/osqueryPullRequestLinux/7046/)) failed one or more tests (**Linux**).

File: osquery/main/fuzz.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  osquery::PluginRequest r;
+  r["action"] = "detach";
+  r["table"] = "file";
+
+  osquery::PluginResponse rsp;
+  osquery::Registry::get().call("sql", r, rsp);
+}
+
+#if 0
```
Comment (obelisk) [inline osquery/main/fuzz.cpp]: Can we have a set of macros at the top that will turn on different fuzz features?

Comment (theopolis) [reply osquery/main/fuzz.cpp]: This was meant as example code. Not really for switching things off and on, but happy to change if you have strong opinions.

**Comment by osqueryer** [comment]
:no_good: The commit 146b3569fbf7c06c17a75b9c706f3521540abb84 (Job results: [3522](https://jenkins.osquery.io/job/osqueryPullRequestAudit/3522/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:-1: The commit 70d01eecaba41d82078316470a6ac0f56fe78f47 (Job results: [7054](https://jenkins.osquery.io/job/osqueryPullRequestLinux/7054/)) failed one or more tests (**Linux**).

**Comment by theopolis** [comment]
I’m going to make some edits to the SDK/FUZZ CMake variables to restore them to the previous logic. I am not happy with the related changes in this PR.

**Comment by osqueryer** [comment]
:-1: The commit efd55efe8ea4ac4f3022338d5c91e1d99bcea286 (Job results: [3940](https://jenkins.osquery.io/job/osqueryPullRequestWindows/3940/)) failed one or more tests (**Windows**).

**Pull request merged**